src/components/PlantingSites/editor/Subzones.tsx

```diff
@@ -34,31 +35,39 @@ export type SubzonesProps = {
   site: PlantingSite;
 };
 
+const featureSiteSubzones = (site: PlantingSite): Record<number, FeatureCollection> =>
+  (site.plantingZones ?? []).reduce(
+    (subzonesMap, zone) => {
+      subzonesMap[zone.id] = {
+        type: 'FeatureCollection',
+        features: zone.plantingSubzones.map(plantingSubzoneToFeature),
+      };
+      return subzonesMap;
+    },
+    {} as Record<number, FeatureCollection>
+  );
+
 export default function Subzones({ onValidate, site }: SubzonesProps): JSX.Element {
   const [selectedZone, setSelectedZone] = useState<number | undefined>();
-  const [zones, setZones] = useState<FeatureCollection | undefined>();
   // map of zone id to subzones
-  const [subzones, setSubzones] = useState<Record<number, FeatureCollection>>({});
+  const [subzones, setSubzones, undo, redo] = useUndoRedoState<Record<number, FeatureCollection>>(
+    featureSiteSubzones(site)
```

Comment:
same here, first value on stack should be the initial subzones state, hence not leveraging a useEffect